GL/texture.c

```diff
@@ -481,6 +490,16 @@ static void _rgba4444_to_argb4444(const GLubyte* source, GLushort* dest) {
     *dest = ((*src & 0x000F) << 12) | *src >> 4;
 }
 
+static void _rgba4444_to_rgba8888(const GLubyte* source, GLushort* dest) {
+    GLushort src = *((GLushort*) source);
+    GLubyte* dst = (GLubyte*) dest;
+
+    dst[0] = (src & 0xF000) * 2;
```

Comment:
This block of conversions is almost certainly not what you want to do, considering the first two elements of dst will always be 0 with what is currently here.

You probably want something like this instead:
`dst[0] = (src >> 8) & 0xf0;
dst[1] = (src >> 4) & 0xf0;
dst[2] = (src & 0xf0);
dst[3] = (src << 4) & 0xf0;`

Reply:
Thanks! I'm refactoring these conversions anyway as the target array is no longer always an array of shorts as indexes could be bytes. Cheers for the review!